dagster_utils/contrib/data_repo/typing.py

```diff
@@ -0,0 +1,10 @@
+"""
+Types and annotations useful for interaction with TDR
+"""
+
+
+class JobId(str):
```

Comment:
This is going to be a common type used everywhere, I thought now is a good time to extract to this lib.